### ISSUE TITLE
Unable to login as regular user

### ISSUE THREAD

**Issue by mattman107** [HuakunShen/wol-web#12]
I am able to login to the PocketBase console `http://localhost:8090/_/` with the superuser credentials I created. 

After I create a regular user with a username and password, I am unable to login to the WOL webpage `http://localhost:8090/auth` using the regular user account. I just get the error below:

![image](https://github.com/user-attachments/assets/e6b967e9-4f24-4810-b208-108acde656af)

I also checked the docker logs but nothing displayed but this:
![image](https://github.com/user-attachments/assets/c5bd543a-a306-4299-a541-6d010c6cc642)

**Comment by Bmv1052** [comment]
I had the same problem that couldn't login as a normal user

**Comment by shengonglin** [comment]
I had the same problem too

**Comment by HuakunShen** [comment]
Can you share the docker command you used? 
Also OS and environment e.g. Windows, Linux, WSL, Mac?
I had no problem creating user and login.
@mattman107 If you can't login as superuser, how can you create a regular user?

**Comment by HuakunShen** [comment]
I will make a instruction video on YouTube tomorrow to demonstrate how to use it.

**Comment by mattman107** [comment]
@HuakunShen 
> Also OS and environment e.g. Windows, Linux, WSL, Mac?

My OS is Truenas Scale Electric Eel (Linux)

> Can you share the docker command you used? 

I created the container using their built in app creation wizard. I will also try to use a compose yaml file to see if that has any effect and will get back to you about it.

> If you can't login as superuser, how can you create a regular user?

I **was** able to login to the PocketBase dashboard as the superuser and create a new user account. I **was not** able to login to the Wake On Lan application with the standard user account. Sorry for not being clearer in the original post.

**Comment by HuakunShen** [comment]
I also have a truenas scale, I will test the image on it today. I have only tested it on mac and ubuntu.

**Comment by mattman107** [comment]
I will also post what options I used in the Truenas Scale app creation wizard. Maybe that will shed some light on if I missed a crucial option somewhere:

 
![image](https://github.com/user-attachments/assets/216471ce-92ef-40dc-9518-0e0e6386baaa)
![image](https://github.com/user-attachments/assets/acd9fe93-ddd4-4fb3-9219-a78c01531347)
![image](https://github.com/user-attachments/assets/99743eae-a585-4394-a3fd-6f1e0b07f8e0)
![image](https://github.com/user-attachments/assets/08733b8f-9016-412a-b820-564bb6f230ee)

**Comment by NotBR** [comment]
I suspect this is to do with running in docker and it somehow trying to post the data to 127.0.0.1. See the javascript console:
`entry.Bls8o69G.js:1 
       POST http://127.0.0.1:8090/api/collections/users/auth-with-password net::ERR_BLOCKED_BY_CLIENT`

**Comment by mattman107** [comment]
> I suspect this is to do with running in docker and it somehow trying to post the data to 127.0.0.1. See the javascript console: `entry.Bls8o69G.js:1 POST http://127.0.0.1:8090/api/collections/users/auth-with-password net::ERR_BLOCKED_BY_CLIENT`

I can confirm my browser is spitting out this same message. 

I'm not a web developer but does this mean the the post request is being sent to my desktop not the server? You would think the post request would want to be sent to the server, but like I said I don't really know what I'm talking about.

![image](https://github.com/user-attachments/assets/90a4636d-c758-43af-b0d1-39ab8c4c150e)

**Comment by NotBR** [comment]
Certainly not the right fix but a horrible hacky workaround is to run this:
```
docker exec wol /bin/sed -i -e 's/http:\/\/127.0.0.1:8090/http:\/\/YOUR.IP.PUT.HERE:8090/g' '/app/pb_public/_app/immutable/chunks/Toaster.svelte_svelte_type_style_lang.DgQw74KS.js'
```

Where wol is the name of the container. Hopefully @HuakunShen can find a proper fix :-)

**Comment by HuakunShen** [comment]
Should be fixed in this PR #13 
Can you try huakunshen/wol:0.3.1?

**Comment by mattman107** [comment]
Will do. I just need to either wait for Truenas to pick up on the image being updated, or figure out how to force update it myself.

**Comment by HuakunShen** [comment]
> Will do. I just need to either wait for Truenas to pick up on the image being updated, or figure out how to force update it myself.

Simply update the image tag to 0.3.1 and save to redeploy.

I tested on my truenas scale, login works perfectly but couldn't wake up my PC. 
I don't see an option to use host network.

**Comment by mattman107** [comment]
Works for me now! I was able to login and wake up a computer!

> I don't see an option to use host network.

I'm wondering if you have a slightly older version? I definitely have a host network option on mine. My Truenas scale version is TrueNAS-24.10.0.2.

**Comment by HuakunShen** [comment]
Yeah I just upgraded to the latest version and got it working with host network.

**Comment by NotBR** [comment]
> @NotBR @mattman107 Thank you for locating the bug, it was due to hard coding 127.0.0.1 as the API url.
> 
> Should be fixed in this PR #13 Can you try image [`huakunshen/wol:0.3.1`](https://hub.docker.com/repository/docker/huakunshen/wol/tags/0.3.1/sha256-4fd9e1e5d69f31f2dad71b2f22310cec1410bbd7e5b80497eb90850f890f0be4)?

Pulled latest docker build:
```
Digest: sha256:d444b166d021d29251b78ab3c338d0c1dcac1ccbc8f7342344c0da186df5a4dc
Status: Downloaded newer image for huakunshen/wol:latest
```

Works without issue now. Thanks.
